packages/core/mesh/teleport/src/muxing/framer.ts

```diff
@@ -24,6 +25,10 @@ export class Framer {
   private _bytesSent = 0;
   private _bytesReceived = 0;
 
+  private _writable = true;
+
+  readonly drain = new Event();
```

Comment:
it would be better to turn this class into two transform streams - they'll handle backpressure automatically

Reply:
I'm not a pro in Nodejs streaming capabilities, but isn't `Duplex` also supposed to handle backpressure automatically?
I _think_ the way it happens in the stream is by checking if the writable buffer is full and emitting the `drain` event when it's ready to accept more data, so one way or another we need to create listeners;
I'm considering merging the fix in that form, since we want to eliminate the OOM caused by queueing callbacks, and do experimentation on Transform streams as a second phase